CHANGES.md

```diff
@@ -1,10 +1,37 @@
+### 0.5.0
+
+- Optimized Michael-Scott queue with a safe and an unsafe versions (@lyrm,
+  @polytypic)
+- Using multicore-magic-dscheck for dscheck tests (@lyrm, @polytypic)
+- Fix cycle dependency in skiplist tests (@lyrm, @polytypic)
+- Require sherlodoc for search bar (@polytypic, @lyrm)
+- Optimize spsc queue : unsafe optimizations (@lyrm, @polytypic)
+- Optimize spsc queue : optimizations without Obj.magic (@lyrm, @polytypic)
+- Add a benchmark using work-stealing deque as a SPMC queue (@polytypic)
+- Fix MS queue test loop with safe point (@polytypic)
+- Upgrade DSCheck to allow testing on OCaml4 with OCaml-CI (@polytypic, @lyrm)
+- Tweak size (@polytypic)
+- Tweak treiber stack (@polytypic)
+- Rewrite and refactor benchmarks using multicore-bench (@polytypic, @lyrm)
+- Add test with OCaml 5.1.1 on Windows and 4.13.x and 4.14.x (@polytypic)
+- Prefer Backoff over Domain.cpu_relax and domain_shims (@polytypic, @Sudha247)
+- Disable implicit transitive dependencies (@polytypic)
+- Fix dune rules to specify package (@polytypic)
+- Replace local backoff module by backoff library (@lyrm, @polytypic)
+- Execute benchmarks in release mode (@polytypic, @Sudha247)
+- Only run Github action on push to main (@polytypic)
+- Lock free skiplist with size (@polytypic, @lyrm, @sooraj-srini)
+- Wait-free size for lock free data structures (@polytypic, @lyrm, @nikochiko)
```

Comment:
Hmm... This seems thorough.

I personally don't usually list internal maintenance (e.g. GH actions changes), fixes/tweaks to previously unreleased features, or (mostly) internal test setup chages, for example, because I assume those are not that useful (i.e. noise) downstream.

Reply:
I'd also mention disabling implicit transitive dependencies.  Otherwise looks good.

Reply:
And the switch to use backoff from separate library.